report/src/main/java/com/megaease/easeagent/report/async/zipkin/AgentByteBoundedQueue.java

```diff
@@ -14,126 +14,103 @@
  * the License.
  */
 
-import java.util.Arrays;
-import java.util.concurrent.locks.Condition;
-import java.util.concurrent.locks.ReentrantLock;
+import lombok.Data;
+
+import java.util.concurrent.LinkedTransferQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * Multi-producer, multi-consumer queue that is bounded by both count and size.
  *
  * <p>This is similar to {@link java.util.concurrent.ArrayBlockingQueue} in implementation.
  */
 public final class AgentByteBoundedQueue<S> implements WithSizeConsumer<S> {
-    final ReentrantLock lock = new ReentrantLock(false);
-    final Condition available = lock.newCondition();
+    private LinkedTransferQueue<DataWrapper<S>> queue = new LinkedTransferQueue<>();
+
+    private final int maxSize;
 
-    final int maxSize;
-    final int maxBytes;
+    private final int maxBytes;
 
-    final S[] elements;
-    final int[] sizesInBytes;
-    int count;
-    int sizeInBytes;
-    int writePos;
-    int readPos;
+    private final AtomicLong sizeInBytes = new AtomicLong(0L);
 
-    @SuppressWarnings("unchecked")
     public AgentByteBoundedQueue(int maxSize, int maxBytes) {
-        this.elements = (S[]) new Object[maxSize];
-        this.sizesInBytes = new int[maxSize];
         this.maxSize = maxSize;
         this.maxBytes = maxBytes;
     }
 
-    /**
-     * Returns true if the element could be added or false if it could not due to its size.
-     */
     @Override
     public boolean offer(S next, int nextSizeInBytes) {
-        lock.lock();
-        try {
-            if (count == maxSize) return false;
-            if (sizeInBytes + nextSizeInBytes > maxBytes) return false;
-
-            elements[writePos] = next;
-            sizesInBytes[writePos++] = nextSizeInBytes;
-
-            if (writePos == maxSize) writePos = 0; // circle back to the front of the array
-
-            count++;
-            sizeInBytes += nextSizeInBytes;
+        if (maxSize == queue.size()) {
+            return false;
+        }
+        if (sizeInBytes.updateAndGet(pre -> pre + nextSizeInBytes) > maxBytes) {
+            return false;
```

Comment:
Add a counter here to record the number of lost, and then reset sizeInBytes please